metricflow/model/validations/measures.py

```diff
@@ -0,0 +1,88 @@
+from collections import defaultdict
+from typing import List, Dict
+from metricflow.instances import MetricModelReference
+from metricflow.model.objects.metric import Metric
+from metricflow.model.objects.user_configured_model import UserConfiguredModel
+from metricflow.model.validations.validator_helpers import (
+    DataSourceElementContext,
+    DataSourceElementReference,
+    DataSourceElementType,
+    FileContext,
+    MetricContext,
+    ModelValidationRule,
+    ValidationIssueType,
+    ValidationError,
+    validate_safely,
+)
+from metricflow.references import MeasureReference
+
+
+class DataSourceMeasuresUniqueRule(ModelValidationRule):
+    """Asserts all measure names are unique across the model."""
+
+    @staticmethod
+    @validate_safely(
+        whats_being_done="running model validation ensuring measures exist in only one configured data source"
+    )
+    def validate_model(model: UserConfiguredModel) -> List[ValidationIssueType]:  # noqa: D
+        issues: List[ValidationIssueType] = []
+
+        measure_references_to_data_sources: Dict[MeasureReference, List] = defaultdict(list)
+        for data_source in model.data_sources:
+            for measure in data_source.measures:
+                if measure.reference in measure_references_to_data_sources:
+                    issues.append(
+                        ValidationError(
+                            context=DataSourceElementContext(
+                                file_context=FileContext.from_metadata(metadata=data_source.metadata),
+                                data_source_element=DataSourceElementReference(
+                                    data_source_name=data_source.name, element_name=measure.name
+                                ),
+                                element_type=DataSourceElementType.MEASURE,
+                            ),
+                            message=f"Found measure with name {measure.name} in multiple data sources with names "
+                            f"({measure_references_to_data_sources[measure.reference]})",
+                        )
+                    )
+                measure_references_to_data_sources[measure.reference].append(data_source.name)
+
+        return issues
+
+
+class MetricMeasuresRule(ModelValidationRule):
```

Comment:
I'm not sure I fully understand why this should be a measure rule instead of a metrics rule, but I'm not sure it's important enough to debate about either 🤷